utils/parser.py

```diff
@@ -390,44 +412,46 @@ async def runner(self, line_of_codes: str) -> AsyncGenerator[Tuple[Any], None]:
         for x in (self.iterator, self.executor):
             await x.__anext__()
 
-    async def handle_repl(self, line: str) -> Union[Tuple[str, Exception], str, int]:
+    async def handle_repl(self, line: str) -> Union[Tuple[str, ReplParserDies], int, bool]:
         try:
             return await self.iterator.asend(line)
         except ReplParserDies as e:
-            lines = traceback.format_exception(type(e), e, e.__traceback__)
+            lines = traceback.format_exc()
             return "".join(lines), e
 
-    async def reading_codeblock(self) -> AsyncGenerator[str, None]:
+    async def reading_codeblock(self) -> AsyncIterator[_ExecIterOutput]:
         codes = itertools.dropwhile(lambda x: x == "", self.codeblock.content.splitlines())
         async for line, no, ex in self.runner(codes):
             if isinstance(indent := await self.handle_repl(line), tuple):
                 _, error = indent
                 indicator = ("...", ">>>")[error.mode]
                 yield f"{indicator} {line}"
                 yield indent
+                return
             number = f"{no} " if self.counter else ""
             compiled = await self.executor.asend((line, indent))
             if ex and indent and compiled:
                 yield compiled
             yield f'{number}{("...", ">>>")[indent]} {line}'
         else:  # eof or raise
             try:
-                if compiled := await self.executor.asend((0, True)):
+                if compiled := await self.executor.asend(("0", True)):
                     yield compiled
-                await self.iterator.asend(0)
+                await self.iterator.asend(None)
             except StopAsyncIteration:
                 return
 
     @staticmethod
     def importer(compiled_str: str, global_vars: Dict[str, Any]) -> str:
-        for ori in re.finditer(IMPORT_REGEX, compiled_str):
-            x = get_import(ori)
-            global_vars.update({x: __import__(x)})
+        def apply_inline_import(m: re.Match[str]) -> str:
+            module_name = get_import(m)
+            global_vars.update({module_name: __import__(module_name)})
+            return module_name
 
-        return re.sub(IMPORT_REGEX, get_import, compiled_str)
+        return re.sub(IMPORT_REGEX, apply_inline_import, compiled_str)
 
     @staticmethod
-    def wrap_function(compiled: str) -> str:
+    def wrap_function(compiled: str) -> Any:
```

Comment:
idk why you put it as Any, this is obviously Union[code, str]

Reply:
`compile` returns Any according to typeshed. Return type is Union[Any, str]